connmgr/connmanager.go

```diff
@@ -68,6 +69,10 @@ type ConnReq struct {
 	conn       net.Conn
 	Addr       net.Addr
 	Permanent  bool
+
+	// Ctx specifies the context to use with the dialer.  The context must
+	// be set or an error will occur.
+	Ctx context.Context
```

Comment:
Per the [documentation for context](https://golang.org/pkg/context/):

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx

Instead, I think a new function named `ConnectContext` should be introduced accordingly in order to avoid breaking the public API as this current implementation does.  As this stands, it's a major break without a major API version bump.

```Go
func (cm *ConnManager) ConnectContext(ctx context.Context, c *ConnReq) {
```